src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableCertificateValidation.cs

```diff
@@ -0,0 +1,162 @@
+                                (IDelegateCreationOperation)operationAnalysisContext.Operation;
+                            
+                            if (delegateCreationOperation.Type == systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol)
+                            {
+                                bool flag = true;
+
+                                var kind = delegateCreationOperation.Target.Kind;
+                                switch (kind)
+                                {
+                                    case OperationKind.AnonymousFunction:
+                                        flag = dealWithAnonmousFunction(delegateCreationOperation.Target);
+                                        break;
+
+                                    case OperationKind.MethodReference:
+                                        IMethodReferenceOperation methodReferenceOperation = (IMethodReferenceOperation)delegateCreationOperation.Target;
+                                        flag = dealWithMethodReference(methodReferenceOperation);
+                                        break;
+                                }
+
+                                if (!flag)
+                                    operationAnalysisContext.ReportDiagnostic(
+                                        Diagnostic.Create(
+                                            Rule,
+                                            delegateCreationOperation.Syntax.GetLocation(),
+                                            kind.ToString()));
+                            }
+                        },
+                        OperationKind.DelegateCreation);
+                });
+        }
+
+        private static bool dealWithAnonmousFunction(IOperation operation)
+        {
+            bool flag = false;
+
+            var descendants = operation.Descendants();
+            foreach (var tmp in descendants)
+            {
+                if (tmp.Kind == OperationKind.Return)
+                {
+                    IReturnOperation a=(IReturnOperation)tmp;
+                    bool returnedValue = (bool)a.ReturnedValue.ConstantValue.Value;
+                    if (returnedValue == false)
+                    {
+                        flag = true;
+                        break;
+                    }
+                }
+            }
+
+            return flag;
+        }
+
+        private static bool dealWithMethodReference(IMethodReferenceOperation methodReferenceOperation)
+        {
+            bool flag = false;
+
+            MethodDeclarationSyntax methodDeclarationSyntax = AnalysisGetStatements(methodReferenceOperation.Method) as MethodDeclarationSyntax;
+            var returnStatementSyntaxs = methodDeclarationSyntax.DescendantNodes().OfType<ReturnStatementSyntax>();
+            foreach (var returnStatementSyntax in returnStatementSyntaxs)
+            {
+                if (returnStatementSyntax.Expression.ToString() == "false")
```

Comment:
This is not recommended. We do not rely on string representation of syntax nodes for semantic checks. For example, this will expression never be true for VB, which uses `False`.

Reply:
Sounds like some VB test cases would be useful here, just to make sure we're not looking at syntax instead of semantics.